tensorflow/compiler/jit/mark_for_compilation_pass_test.cc

```diff
@@ -1677,5 +1677,34 @@ TEST(XlaCompilationTest, IterationIncrementAndGroupDeps) {
   EXPECT_EQ(clusters["some_ctrl_input"], clusters["matmul_0"]);
 }
 
+TEST(XlaCompilationTest, DontClusterTheSpecialIdentityDrivingConstsInLoop) {
+  Scope root = Scope::NewRootScope().ExitOnError();
```

Comment:
The test constructs an incomplete (and somehow not well-formed) graph, but it is enough to test the added feature. Let me know if you have concerns about the incomplete graph. I did not create a full graph because it feels heavy to me.

Reply:
> somehow not well-formed

What about this graph is not well formed?

Reply:
May not be too relevant.

I was thinking that it has a LoopCond but lacks frame info and Enter/Exit/NextIteration, etc.

Reply:
Thanks for taking a look of it. Comments added.